src/htsjdk/java/htsjdk/samtools/BAMIndexMerger.java

```diff
@@ -13,10 +14,65 @@
  * Merges BAM index files for (headerless) parts of a BAM file into a single
  * index file. The index files must have been produced using {@link BAMIndexer2}.
  */
-public class BAMIndexMerger {
+public class BAMIndexMerger implements Closeable {
 
   private static final int UNINITIALIZED_WINDOW = -1;
 
+  private final OutputStream out;
+  private final List<Long> partLengths;
+  private int numReferences;
+  private List<List<BAMIndexContent>> content;
+  private long noCoordinateCount;
+
+  public BAMIndexMerger(final OutputStream out, final long headerLength) {
+    this.out = out;
+    this.partLengths = new ArrayList<>();
+    this.partLengths.add(headerLength);
+  }
+
+  public void processIndex(AbstractBAMFileIndex index, long partLength) {
+    this.partLengths.add(partLength);
+    if (content == null) {
```

Comment:
What is the benefit of using `null` and null checks throughout instead of an empty `List`, which can be final?

Reply:
Good point. I've changed to use a final List.